src/app/public/modules/progress-indicator/progress-indicator.component.ts

```diff
@@ -1,110 +1,138 @@
 import {
   AfterContentInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
   Component,
   ContentChildren,
   EventEmitter,
   Input,
   OnDestroy,
-  Optional,
+  OnInit,
   Output,
   QueryList
 } from '@angular/core';
 
+import {
+  SkyAppWindowRef
+} from '@skyux/core';
+
 import {
   Subject
 } from 'rxjs/Subject';
 
+import 'rxjs/add/operator/takeUntil';
+
 import {
-  SkyWindowRefService
-} from '@skyux/core';
+  SkyProgressIndicatorItemComponent
+} from './progress-indicator-item/progress-indicator-item.component';
 
 import {
   SkyProgressIndicatorChange,
+  SkyProgressIndicatorDisplayMode,
+  SkyProgressIndicatorMessage,
   SkyProgressIndicatorMessageType
 } from './types';
 
-import {
-  SkyProgressIndicatorItemComponent
-} from './progress-indicator-item';
-
-import {
-  SkyProgressIndicatorDisplayMode
-} from './types/progress-indicator-mode';
-
 @Component({
   selector: 'sky-progress-indicator',
   templateUrl: './progress-indicator.component.html',
-  styleUrls: ['./progress-indicator.component.scss']
+  styleUrls: ['./progress-indicator.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class SkyProgressIndicatorComponent implements AfterContentInit, OnDestroy {
-
-  @Input()
-  public startingIndex: number;
-
+export class SkyProgressIndicatorComponent implements OnInit, AfterContentInit, OnDestroy {
   @Input()
-  public get displayMode(): SkyProgressIndicatorDisplayMode {
-    return this._displayMode || SkyProgressIndicatorDisplayMode.Vertical;
-  }
   public set displayMode(value: SkyProgressIndicatorDisplayMode) {
     this._displayMode = value;
   }
 
+  public get displayMode(): SkyProgressIndicatorDisplayMode {
+    if (this._displayMode === undefined) {
```

Comment:
Because the first value of `SkyProgressIndicatorDisplayMode` is zero (it's a numeric `enum`), and the original short circuit evaluated `0` to false, instead of a real value.

Reply:
Makes sense. Thanks!